src/cargo/util/toml/mod.rs

```diff
@@ -2240,13 +2264,16 @@ impl<P: ResolveToPath + Clone> TomlDependency<P> {
             TomlDependency::Workspace(w) => w.optional.unwrap_or(false),
         }
     }
+}
 
+impl TomlDependency {
```

Comment:
Why the new block?

Reply:
Same reasons as for `DetailedTomlDependency`